packages/@aws-cdk/aws-ec2/lib/vpc-peering-connection.ts

```diff
@@ -0,0 +1,132 @@
+import * as iam from '@aws-cdk/aws-iam';
+import * as cdk from '@aws-cdk/core';
+import { CfnVPCPeeringConnection } from './ec2.generated';
+import { IVpc } from './vpc';
+
+/**
+ * VPC with which you are creating the VPC peering connection.
+ */
+export class PeerVpc {
+  constructor(
+    /**
+     * The ID of the VPC with which you are creating the VPC peering connection
+     */
+    public readonly vpcId: string,
+
+    /**
+     * The Region code for the accepter VPC
+     *
+     * @default - The Region in which you make the request
+     */
+    public readonly region?: string,
+
+    /**
+     * The AWS account ID of the owner of the accepter VPC
+     *
+     * @default - Your AWS account ID
+     */
+    public readonly account?: string,
+  ) {}
+}
+
+/**
+ * Options for constructing VpcPeeringConnection
+ */
+export interface VpcPeeringConnectionOptions {
+  /**
+   * The name of VPC peering connections
+   *
+   * @default - generated by CloudFormation
+   */
+  readonly vpcPeeringConnectionName?: string
+
+  /**
+   * VPC with which you are creating the VPC peering connection.
+   */
+  readonly peerVpc: PeerVpc
+
+  /**
+   * The role, which is required when you are peering a VPC in a different AWS account.
+   *
+   * @default - conditional
+   */
+  readonly role?: iam.IRole
```

Comment:
I guess we could probably create this role? What are the permissions on it that are necessary? Please give links to documentation? 

The current documentation is not really enough for people to make use of this feature, and we should always be looking if we can simplify it.

Reply:
The peer(acceptor)'s acceptance is required to establish vpc peering connection.

This is usually done by hand on the console.

![image](https://user-images.githubusercontent.com/15084045/89972023-c1558080-dc97-11ea-9de9-07dbefa4e787.png)

Maybe an IAM Role with this policy is needed:

```
"Effect": "Allow",
"Action": [
  "ec2:AcceptVpcPeeringConnection"
],
"Resource": [
  "arn:aws:ec2:(requestor-region):(requestor-account):vpc-peering-connection/(vpc-peering-connection-id)",
  "arn:aws:ec2:(acceptor-region):(acceptor-account):vpc/(vpc-id)",
]
```

Reply:
I'm not sure it's correct. I'll look into it a little bit more.

Reply:
> https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-vpcpeeringconnection.html#cfn-ec2-vpcpeeringconnection-peerrolearn